src/index.ts

```diff
@@ -80,28 +88,19 @@ export const compare = memoize((versionA: string | null, versionB: string | null
 		return -1;
 	}
 	if (!isAValid && !isBValid) {
-		if (versionA > versionB) {
-			return 1;
-		}
-		if (versionA < versionB) {
-			return -1;
-		}
-		return 0;
+		return compareValues(versionA, versionB);
 	}
+
 	const semverResult = semver.compare(versionA, versionB);
 	if (semverResult !== 0) {
 		return semverResult;
 	}
-	const revA = getRev(versionA);
-	const revB = getRev(versionB);
-	if (revA !== revB) {
-		return revA > revB ? 1 : -1;
-	}
-	const devA = Number(isDevelopmentVersion(versionA));
-	const devB = Number(isDevelopmentVersion(versionB));
-	if (devA !== devB) {
-		return devA > devB ? -1 : 1;
```

Comment:
It looks like we've removed the dev version comparison to fall back to `localeCompare`, is that right? I think this is a bug, and we still need to do a `isDevelopmentVersion` comparison of some sort before we fall back to locale comparison.

With just `+dev` vs `+prod` it doesn't make a difference, but it does matter if there's other things in the build metadata. `''` should be greater than `'dev'` (though I think maybe this is still an open question?), `+123asd.prod` vs `+asd123.dev` should really be sorted primarily by dev/prod, ignoring any other parts, which could be git hashes or something, `+rev1.dev` and `+prod.rev1` should sort `prod` above `dev`, ignoring the order of these metadata parts.

I think this just means we need to check for dev status (since this is normalized and valid by this point, I'd just parse and look for `dev` in `parsedVersion.build`), and sort dev < all non-dev.